modules/version-checker/version_checker.go

```diff
@@ -0,0 +1,202 @@
+package version_checker
+
+import (
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/shell"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/hashicorp/go-version"
+	"github.com/stretchr/testify/require"
+	"regexp"
+)
+
+// VersionCheckerBinary is an enum for supported version checking.
+type VersionCheckerBinary int
+
+// List of binaries supported for version checking.
+const (
+	Docker VersionCheckerBinary = iota
+	Terraform
+	Packer
+)
+
+// VersionCheckerType is an enum for different type of version checking available.
+type VersionCheckerType int
+
+const (
+	MinimumVersion VersionCheckerType = iota
+	VersionConstraint
+)
+
+const (
+	// versionRegexMatcher is a regex used to extract version string from shell command output.
+	versionRegexMatcher = `\d+(\.\d+)+`
+	// defaultVersionArg is a default arg to pass in to get version output from shell command.
+	defaultVersionArg = "--version"
+)
+
+type CheckVersionParams struct {
+	// BinaryPath is a path to the binary you want to check the version for.
+	BinaryPath string
+	// Binary is the name of the binary you want to check the version for.
+	Binary VersionCheckerBinary
+	// VersionCheckerType to specify different ways to enforce certain version on binary.
+	VersionCheckerType VersionCheckerType
+	// MinimumVersion is a string literal containing a minimum version (e.g., 1.2.1)
+	MinimumVersion string
+	// VersionConstraint is a string literal containing one or more conditions, which are separated by commas.
+	// More information here:https://www.terraform.io/language/expressions/version-constraints
+	VersionConstraint string
+	// WorkingDir is a directory you want to run the shell command.
+	WorkingDir string
+}
+
+// CheckVersionE checks whether the given Binary version is greater than or equal
+// to the given expected version.
+func CheckVersionE(
+	t testing.TestingT,
+	params CheckVersionParams) error {
+
+	if err := validateParams(params); err != nil {
+		return err
+	}
+
+	binaryVersion, err := getVersionWithShellCommand(t, params)
+	if err != nil {
+		return err
+	}
+
+	var versionConstraint string
+	switch params.VersionCheckerType {
+	case MinimumVersion:
+		versionConstraint = ">" + params.MinimumVersion
+	case VersionConstraint:
+		versionConstraint = params.VersionConstraint
```

Comment:
Ah what I actually meant was to remove the type altogether from the struct and only support the version constraint interface. It makes for a better UX because it removes one more thing the user has to configure.

Does that make sense?

Reply:
Ah got it. updated.